File: day01/src/main/java/com/example/day01/SecondActivity.java
package com.example.day01;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

import com.example.mylibrary.TipUtils;

public class SecondActivity extends AppCompatActivity {

    private EditText et_second_msg;
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        intent = this.getIntent();

        et_second_msg = findViewById(R.id.et_second_msg);
        String msg = intent.getStringExtra("msg").toString();
        et_second_msg.setText(msg);

    }

    // activity_second.xml 中两个按钮的onClick中的回调方法，在此重写
    public void return1(View view) {
        TipUtils.shortTips(this, "一般返回");
        // 关闭Activity
        this.finish();
    }

    public void return2(View view) {
        TipUtils.shortTips(this, "带回调返回");

        String msgReturn = et_second_msg.getText().toString();
        intent.putExtra("msgReturn", msgReturn);

        int resultCode = 1;
        setResult(resultCode, intent);
        // 关闭Activity
        this.finish();
    }
}

File: day01k/src/main/java/com/example/day01k/SecondActivity.kt
package com.example.day01k

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_second.*

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        // 初始化et_second_msg
        et_second_msg.setText(intent.getStringExtra("msg").toString())

    }

    fun return1(view: View) {
        finish()
    }

    // 带结果的返回
    fun return2(view: View) {
        intent.putExtra("msg", et_second_msg.text.toString())
        setResult(1,intent)
        finish()
    }
}